lib/report_generators/search_term_by_dimensions.rb

```diff
@@ -0,0 +1,275 @@
+    end
+
+  end
+
+  def upload_to_drive
+    drive_client = @google_api_client.drive_client
+
+   # Upload a file
+    metadata = GoogleApiClient::DRIVE_FILE.new(name: self.report_basename, mime_type: 'application/vnd.google-apps.spreadsheet')
+    file = drive_client.create_file(metadata, upload_source: self.report_output_path, content_type: 'text/csv', supports_team_drives: true)
+    drive_client.update_file(file.id, add_parents: @google_parent_id)
+
+    filter_spreadsheet(file)
+  end
+
+  def filter_spreadsheet(file)
+    sheets_client = @google_api_client.sheets_client
+    spreadsheet = sheets_client.get_spreadsheet(file.id)
+
+    requests = {
+      requests: [
+        {add_filter_view: {
+          filter: {
+            title: 'all query summary',
+            range: {
+              sheet_id: spreadsheet.sheets[0].properties.sheet_id
+            },
+            criteria: {
+              '2': {
+                hidden_values: ['encore', 'drupalsearch', 'betasearch', 'beta search', 'catalog', 'sitesearch']
+              }
+            }
+          }
+        }},
+        {add_filter_view: {
+          filter: {
+            title: 'encore — header search vs browse',
+            range: {
+              sheet_id: spreadsheet.sheets[0].properties.sheet_id
+            },
+            criteria: {
+              '2': {
+                hidden_values: ['all', 'drupalsearch', 'betasearch', 'beta search', 'catalog', 'sitesearch']
+              },
+              '3': {
+                hidden_values: ['all', 'unknown']
+              }
+            }
+          }
+        }},
+        {add_filter_view: {
+          filter: {
+            title: 'header search',
+            range: {
+              sheet_id: spreadsheet.sheets[0].properties.sheet_id
+            },
+            criteria: {
+              '2': {
+                hidden_values: ['all', 'catalog']
+              },
+              '3': {
+                hidden_values: ['all', 'encoresearchform', 'unknown']
+              }
+            }
+          }
+        }},
+        {add_filter_view: {
+          filter: {
+            title: 'unknown where searched from',
+            range: {
+              sheet_id: spreadsheet.sheets[0].properties.sheet_id
+            },
+            criteria: {
+              '3': {
+                hidden_values: ['all', 'headersearch', 'encoresearchform']
+              }
+            }
+          }
+        }}
+      ]
+    }
+
+    sheets_client.batch_update_spreadsheet(file.id, requests, {})
+  end
+
+end
+
+class TermEventsProcessor
+  attr_accessor :term, :query_segments, :click_segments
+
+  @@dimensions = CONFIG[:dimensions]
+
+  def initialize(term:, query_segments:, click_segments:)
+
+    @term = term
+    @query_segments = query_segments
+    @click_segments = click_segments
+
+  end
+
+  def process
+    process_data_for_dimensions(@@dimensions.dup)
+  end
+
+  def process_data_for_dimensions(dimensions, values: {})
```

Comment:
Hey @nodanaonlyzuul! This method is the core of the changes — everything else is just reorganizing to support this.